services/tasks-svc/internal/patient/projections/patient_postgres_projection/patient_postgres_projection.go

```diff
@@ -0,0 +1,185 @@
+package patient_postgres_projection
+
+import (
+	"context"
+	"fmt"
+	"github.com/EventStore/EventStore-Client-Go/esdb"
+	"github.com/google/uuid"
+	"github.com/jackc/pgx/v5/pgtype"
+	zlog "github.com/rs/zerolog/log"
+	"hwdb"
+	"hwes"
+	"hwes/eventstoredb/projections/custom"
+	"hwutil"
+	patientEventsV1 "tasks-svc/internal/patient/events/v1"
+	"tasks-svc/repos/patient_repo"
+)
+
+type Projection struct {
+	*custom.CustomProjection
+}
+
+func NewProjection(es *esdb.Client, serviceName string) *Projection {
+	subscriptionGroupName := fmt.Sprintf("%s-postgres-projection", serviceName)
+	p := &Projection{custom.NewCustomProjection(es, subscriptionGroupName)}
+	p.initEventListeners()
+	return p
+}
+
+func (p *Projection) initEventListeners() {
+	p.RegisterEventListener(patientEventsV1.PatientCreated, p.onPatientCreated)
+	p.RegisterEventListener(patientEventsV1.BedAssigned, p.onBedAssigned)
+	p.RegisterEventListener(patientEventsV1.BedUnsassigned, p.onBedUnassigned)
+	p.RegisterEventListener(patientEventsV1.PatientDischarged, p.onPatientDischarged)
+	p.RegisterEventListener(patientEventsV1.NotesUpdated, p.onNotesUpdated)
+	p.RegisterEventListener(patientEventsV1.HumanReadableIdentifierUpdated, p.onHumanReadableIdentifierUpdated)
+	p.RegisterEventListener(patientEventsV1.PatientReadmitted, p.onPatientReadmitted)
+}
+
+// Event handlers
+func (a *Projection) onPatientCreated(ctx context.Context, evt hwes.Event) (error, esdb.Nack_Action) {
+	log := zlog.Ctx(ctx)
+	patientRepo := patient_repo.New(hwdb.GetDB())
+
+	var payload patientEventsV1.PatientCreatedEvent
+	if err := evt.GetJsonData(&payload); err != nil {
+		log.Error().Err(err).Msg("unmarshal failed")
+		return err, esdb.Nack_Retry
```

Comment:
when the payload is invalid, should we retry? I'm not sure about this either way

Reply:
I was not quite sure about this one either. I've talked with @MaxSchaefer about the `NackActions` and we decided retries are sufficient for our current state of `tasks-svc`